app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -0,0 +1,185 @@
+import Button from "components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "components/Dialog";
+import {
+  HOST_REQUEST_REFERENCE_EXPLANATION,
+  HOST_REQUEST_REFERENCE_SUCCESS_DIALOG,
+  OKAY,
+  REFERENCE_SUCCESS,
+  SUBMIT,
+} from "features/profile/constants";
+import {
+  useWriteFriendReference,
+  useWriteHostReference,
+} from "features/profile/hooks/referencesHooks";
+import ReferenceOverview from "features/profile/view/leaveReference/formSteps/submit/ReferenceOverview";
+import {
+  ReferenceContextFormData,
+  useReferenceStyles,
+} from "features/profile/view/leaveReference/ReferenceForm";
+import { User } from "pb/api_pb";
+import { ReferenceType } from "pb/references_pb";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useHistory, useParams } from "react-router";
+import { baseRoute, referenceTypeRoute, userRoute } from "routes";
+import {
+  WriteFriendReferenceInput,
+  WriteHostRequestReferenceInput,
+} from "service/references";
+
+export interface SubmitReferenceProps {
+  user: User.AsObject;
+  referenceData: ReferenceContextFormData;
+}
+
+export default function SubmitReference({
+  user,
+  referenceData,
+}: SubmitReferenceProps) {
+  const {
+    writeFriendReference,
+    reset: resetFriendReferenceWriting,
+    error: friendReferenceError,
+    isLoading: isFriendReferenceLoading,
+  } = useWriteFriendReference(user.userId);
+  const {
+    writeHostRequestReference,
+    reset: resetHostRequestReferenceWriting,
+    error: hostRequestReferenceError,
+    isLoading: isHostRequestReferenceLoading,
+  } = useWriteHostReference(user.userId);
+
+  const { referenceType, hostRequest } = useParams<{
+    referenceType: string;
+    hostRequest?: string;
+  }>();
+  const theme = useTheme();
+  const classes = useReferenceStyles();
+  const history = useHistory();
+  const [isOpen, setIsOpen] = useState(false);
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { handleSubmit } = useForm<ReferenceContextFormData>();
+
+  const onFriendReferenceSubmit = () => {
+    const formData: WriteFriendReferenceInput =
+      referenceData.wasAppropriate === "true"
+        ? {
+            toUserId: user.userId,
+            wasAppropriate: true,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          }
+        : {
+            toUserId: user.userId,
+            wasAppropriate: false,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          };
+
+    resetFriendReferenceWriting();
+    writeFriendReference(
+      {
+        referenceData: formData,
+      },
+      {
+        onSuccess: () => {
+          history.push(`${userRoute}/${user.username}/references`);
+        },
+      }
+    );
+    window.scroll({ top: 0 });
+  };
+
+  const onHostReferenceSubmit = () => {
+    if (hostRequest && +hostRequest !== undefined) {
```

Comment:
```suggestion
    if (hostRequest) {
```
This alone should be enough since `hostRequest` would be falsey if it's undefined

Reply:
Is it alright to keep `&& +hostRequest` too, since if the `hostRequest` string is not a number it will become `NaN`, or should we trust that the `hostRequest` param is a valid string now that we have the routing set up better?

Reply:
in that case, probably better to use the `isNaN` function for this exact purpose and do `if (!isNaN(hostRequestId)) { ... }` -